zipline/gens/tradesimulation.py

```diff
@@ -266,11 +268,14 @@ def _call_handle_data(self):
         Call the user's handle_data, returning any orders placed by the algo
         during the call.
         """
+        time_start = utcnow()
         self.algo.event_manager.handle_data(
             self.algo,
             self.current_data,
             self.simulation_dt,
         )
+        time_end = utcnow()
+        self.algo.call_handle_data_duration.append((time_end-time_start).total_seconds()*1000)
```

Comment:
It would make it more feasible.

I would still lean towards putting it in call_algorithm_handle_data, so that those who aren't using this metadata don't need to run it every bar, and also to leave tradesimulation as focused as possible on getting the trade data in the right places.